### PR TITLE
tests: move IT in `executor` to `tests/integrationtest` (PART 10)

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: ref #47076

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by Defined2014** [comment]
@hawkingrei @YangKeao @SeaRise  PTAL

**Review by hawkingrei — APPROVED** [review]

File: tests/integrationtest/t/executor/jointest/join.test [hunk 1]
```diff
@@ -586,3 +586,473 @@ insert into t3 values (1, 'A');
...
+create table t2(fa int, fb int);
+insert into t1 values (1,1),(1,1),(1,2),(1,2),(1,2),(1,3);
+insert into t2 values (1,1),(1,2),(1,3);
+select f1,f2 from t1 group by f1,f2 having count(1) >= all (select fb from t2 where fa = f1);
+
+DROP TABLE IF EXISTS t1, t2;
+CREATE TABLE t1(a INT);
+CREATE TABLE t2 (d BINARY(2), PRIMARY KEY (d(1)), UNIQUE KEY (d));
+INSERT INTO t1 values(1);
+SELECT 1 FROM executor__jointest__join.t1, executor__jointest__join.t2 WHERE 1 = (SELECT executor__jointest__join.t2.d FROM executor__jointest__join.t2 WHERE executor__jointest__join.t1.a >= 1) and executor__jointest__join.t2.d = 1;
```
Comment (SeaRise) [inline tests/integrationtest/t/executor/jointest/join.test:987]:
```suggestion
SELECT 1 FROM t1, t2 WHERE 1 = (SELECT t2.d FROM t2 WHERE t1.a >= 1) and t2.d = 1;
```

**Review by SeaRise — APPROVED** [review]
lgtm

**Pull request merged**